src/web/pages/roles/RoleComponent.tsx

```diff
@@ -21,7 +29,41 @@ import {
   loadAllEntities as loadAllUsers,
   selector as userSelector,
 } from 'web/store/entities/users';
-import PropTypes from 'web/utils/PropTypes';
+
+interface CreatePermissionData {
+  roleId: string;
+  name: string;
+}
+
+interface CreateSuperPermissionData {
+  roleId: string;
+  groupId: string;
+}
+
+interface DeletePermissionData {
+  roleId: string;
+  permissionId: string;
+}
+
+interface RoleComponentProps {
+  children: (props: {
+    create: (role?: Role) => void;
+    edit: (role: Role) => void;
+    clone: (role: Role) => void;
+    delete: (role: Role) => Promise<void>;
+    download: (role: Role) => void;
+  }) => React.ReactNode;
+  onCloneError?: (error: Rejection) => void;
+  onCloned?: GotoDetailsFunc;
+  onCreateError?: (error: Rejection) => void;
+  onCreated?: GotoDetailsFunc;
+  onDeleteError?: (error: Rejection) => void;
+  onDeleted?: () => void;
+  onDownloadError?: (error: Rejection) => void;
+  onDownloaded?: OnDownloadedFunc;
+  onSaveError?: (error: Rejection) => void;
+  onSaved?: (response: Response<ActionResult, XmlMeta>) => void;
```

Comment:
I am going to change this to void in all commands. The actions result just contains the id of the change entity and two messages with some ok status. So it's irrelevant for the save use case.

```suggestion
  onSaved?: () => void;
```